pkg/combustion/kubernetes_test.go

```diff
@@ -821,3 +1511,191 @@ func TestConfigureKubernetes_SuccessfulRKE2ServerWithManifests(t *testing.T) {
 	assert.Contains(t, contents, "name: my-nginx")
 	assert.Contains(t, contents, "image: nginx:1.14.2")
 }
+
+func TestKubernetesVIPManifestValidIPV4(t *testing.T) {
+	k8s := &image.Kubernetes{
+		Version: "v1.30.3+rke2r1",
+		Network: image.Network{
+			APIVIP4: "192.168.1.1",
+		},
+	}
+
+	manifest, err := kubernetesVIPManifest(k8s)
+	require.NoError(t, err)
+
+	assert.Contains(t, manifest, "- 192.168.1.1/32")
+	assert.Contains(t, manifest, "- name: rke2-api")
+	assert.NotContains(t, manifest, "ipFamilies:\n      - IPv6")
+	assert.NotContains(t, manifest, "ipFamilyPolicy: SingleStack")
+}
+
+func TestKubernetesVIPManifestValidIPV6(t *testing.T) {
+	k8s := &image.Kubernetes{
+		Version: "v1.30.3+k3s1",
+		Network: image.Network{
+			APIVIP6: "fd12:3456:789a::21",
+		},
+	}
+
+	manifest, err := kubernetesVIPManifest(k8s)
+	require.NoError(t, err)
+
+	assert.Contains(t, manifest, "- fd12:3456:789a::21/128")
+	assert.Contains(t, manifest, "- name: k8s-api")
+	assert.Contains(t, manifest, "ipFamilies:\n    - IPv6")
+	assert.Contains(t, manifest, "ipFamilyPolicy: SingleStack")
+	assert.NotContains(t, manifest, "rke2")
+}
+
+func TestKubernetesVIPManifestDualstack(t *testing.T) {
+	k8s := &image.Kubernetes{
+		Version: "v1.30.3+k3s1",
+		Network: image.Network{
+			APIVIP4: "192.168.1.1",
+			APIVIP6: "fd12:3456:789a::21",
+		},
+	}
+
+	manifest, err := kubernetesVIPManifest(k8s)
+	require.NoError(t, err)
+
+	assert.Contains(t, manifest, "- 192.168.1.1/32")
+	assert.Contains(t, manifest, "- fd12:3456:789a::21/128")
+	assert.Contains(t, manifest, "- name: k8s-api")
+	assert.NotContains(t, manifest, "ipFamilies:\n      - IPv6")
+	assert.NotContains(t, manifest, "ipFamilyPolicy: SingleStack")
+}
+
+func TestCreateNodeIPScriptDualstackIPv6PrioK3s(t *testing.T) {
```

Comment:
Nit: Such names are very hard to read. We can use underscores as many times as necessary to make it better. E.g.
```suggestion
func TestCreateNodeIPScript_DualStack_IPv6Prio_K3s(t *testing.T) {
```